### PR TITLE
feat: add user and session data to next and react

### PR DESCRIPTION
**Pull request by Ashutosh-Bhadauriya**
Adds getting user and session data to nextjs and react guides

### CODE REVIEW

File: quickstarts/frontend/react.mdx [hunk 1]
```diff
@@ -301,12 +301,157 @@ You can use `@teamhanko/hanko-elements` to easily manage user logouts. Below, we
...
+  return sessionState;
+}
+```
+</Tab>
+</Tabs>
+
+This is how you can use them according to your needs.
+
+```tsx components/userData.tsx
+import dynamic from 'next/dynamic';
```
Comment (lfleischmann) [inline quickstarts/frontend/react.mdx]: Do we really want to showcase usage of `next` packages in the `react` guide?

File: quickstarts/frontend/react.mdx [hunk 2]
```diff
@@ -301,12 +301,157 @@ You can use `@teamhanko/hanko-elements` to easily manage user logouts. Below, we
...
 
 You can customize the appearance of `hanko-auth` and `hanko-profile` components using CSS variables and parts. Refer to our [customization guide](/guides/hanko-elements/customize-appearance).
 
+## Getting User and Session Data
+
+### Client Side
+
+We will create two custom hooks to fetch the current user's data and session data from `@teamhanko/hanko-elements`. The `useUserData` hook leverages the `hanko.user.getCurrent()` method to retrieve the currently logged-in user's data. 
+
+On the other hand, the `useSessionData` hook uses `hanko.session.isValid()` and `hanko.session.get()` methods from Hanko to validate and fetch the current session data.
```
Comment (lfleischmann) [inline quickstarts/frontend/react.mdx]:
```suggestion
On the other hand, the `useSessionData` hook uses the `hanko.session.isValid()` and `hanko.session.get()` methods to validate and fetch the current session data.
```

File: quickstarts/frontend/react.mdx [hunk 3]
```diff
@@ -301,12 +301,157 @@ You can use `@teamhanko/hanko-elements` to easily manage user logouts. Below, we
...
-
 ## Customize component styles
 
 You can customize the appearance of `hanko-auth` and `hanko-profile` components using CSS variables and parts. Refer to our [customization guide](/guides/hanko-elements/customize-appearance).
 
+## Getting User and Session Data
+
+### Client Side
+
+We will create two custom hooks to fetch the current user's data and session data from `@teamhanko/hanko-elements`. The `useUserData` hook leverages the `hanko.user.getCurrent()` method to retrieve the currently logged-in user's data.
```
Comment (lfleischmann) [inline quickstarts/frontend/react.mdx]:
```suggestion
We will create two custom hooks to fetch the current user's data and session data using `@teamhanko/hanko-elements`. The `useUserData` hook leverages the `hanko.user.getCurrent()` method to retrieve the currently logged-in user's data. 
```

File: quickstarts/fullstack/next.mdx [hunk 4]
```diff
@@ -410,9 +409,236 @@ export const config = {
...
+</Tabs>
+
+### Server Side
+
+On the server side, you can extract the `userID` from the JWT. However, if you need to retrieve the complete user details, you'll have to utilize the [Hanko Admin API](https://docs.hanko.io/api-reference/admin/status/status-page).
+
+The Hanko Admin API provides detailed information about the status, user management, metrics and more. To get data for a specific user, you'll need to call the `/users/{id}` endpoint, where id is the user id previously obtained from the JWT.
+
+<Info>
+Hanko Admin API is a paid feature and must be enabled separately. You also need an API key `secret` to access the Hanko Admin API. You can generate one under the `Settings/API Keys` section of your project.
```
Comment (lfleischmann) [inline quickstarts/fullstack/next.mdx]:
```suggestion
The Hanko Admin API is a paid feature and must be enabled explicitly. You also need an API key `secret` to access the Hanko Admin API. You can generate one under the `Settings/API Keys` section of your project.
```

File: quickstarts/fullstack/next.mdx [hunk 5]
```diff
@@ -410,9 +409,236 @@ export const config = {
...
+};
+
+export default UserData;
+```
+</Tab>
+</Tabs>
+
+### Server Side
+
+On the server side, you can extract the `userID` from the JWT. However, if you need to retrieve the complete user details, you'll have to utilize the [Hanko Admin API](https://docs.hanko.io/api-reference/admin/status/status-page).
```
Comment (lfleischmann) [inline quickstarts/fullstack/next.mdx]: The  link leads to the status page but should probably lead to https://docs.hanko.io/api-reference/admin/user-management/get-a-user-by-id.

File: quickstarts/fullstack/next.mdx [hunk 6]
```diff
@@ -410,9 +409,236 @@ export const config = {
...
+
+The Hanko Admin API provides detailed information about the status, user management, metrics and more. To get data for a specific user, you'll need to call the `/users/{id}` endpoint, where id is the user id previously obtained from the JWT.
+
+<Info>
+Hanko Admin API is a paid feature and must be enabled separately. You also need an API key `secret` to access the Hanko Admin API. You can generate one under the `Settings/API Keys` section of your project.
+Keep in mind, that the API key `secret` is only showed once, so make sure to store it in a safe place, like an environmental variable as shown in the following example.
```
Comment (lfleischmann) [inline quickstarts/fullstack/next.mdx]:
```suggestion
Keep in mind, that the API key `secret` is only shown once, so make sure to store it in a safe place, like an environmental variable as shown in the following example.
```

File: quickstarts/frontend/react.mdx (hunk 1 above)
Comment (Ashutosh-Bhadauriya) [reply quickstarts/frontend/react.mdx]: ahh forgot to change this, will fix

File: quickstarts/frontend/react.mdx [hunk 7]
```diff
@@ -301,12 +301,157 @@ You can use `@teamhanko/hanko-elements` to easily manage user logouts. Below, we
...
+  const [hanko, setHanko] = useState<Hanko>();
+  const [userState, setUserState] = useState<HankoUser>({
+    id: "",
+    email: "",
+    loading: true,
+    error: null,
+  });
+
+  useEffect(() => {
+    import("@teamhanko/hanko-elements").then(({ Hanko }) =>
```
Comment (lfleischmann) [inline quickstarts/frontend/react.mdx]: Do we really need a dynamic import in react? If I remember correctly, none of our "old" examples" do this/"old" doc guides for react mention this.

**Review by lfleischmann — APPROVED** [review]

**Pull request merged**